daemon/k8s_watcher.go

```diff
@@ -289,7 +289,19 @@ func (d *Daemon) EnableK8sWatcher(reSyncPeriod time.Duration) error {
 				},
 			},
 		)
+		d.k8sResourceSyncWaitGroup.Add(1)
 		go policyController.Run(wait.NeverStop)
+		go func() {
+			completed := make(<-chan struct{})
```

Comment:
I think both `completed := make(<-chan struct{})` should be replaced with a hard timeout, otherwise this might hang forever until we receive the network policies from kube-apiserver.

cc @tgraf what should be the expected behavior? Also, if kube-apiserver is not available Cilium will not start and kubelet will kill it because of the liveliness probe.

Reply:
How can the creation of the channel block? I agree that the waiting on `k8sResourceSyncWaitGroup` should have a timeout.

Reply:
@tgraf `cache.WaitForCacheSync` in the next line will keep retrying forever as the `completed` channel is never closed.

Reply:
Agreed but this is done in the background and we would wait later on for all caches that need to be synced and the timeout there will kill the daemon.

Reply:
Timeout will be there in next upload . This channel is there for the sake of providing it to `WaitForCacheSync` and nothing more - we could potentially use it later like it is done in the Kubernetes example for controllers: https://github.com/kubernetes/kubernetes/blob/5de3896d0d18f072664650b94a8f6089be420cc0/staging/src/k8s.io/sample-controller/controller.go#L152-L176